tests/lax_control_flow_test.py

```diff
@@ -2739,5 +2740,30 @@ def side_effecting_scan(carry, val):
       lax.scan(side_effecting_scan, None, jnp.ones((2, 2)))
       lst[0] += 1
 
+  def test_while_loop_fixed_point_with_nested_named_axes(self):
+    def f(x):
+      z = x + lax.axis_index('a')
+      y = x + lax.axis_index('b')
+      def cond(carry):
+        i, x = carry
+        return x < 5
+      def body(carry):
+        i, x = carry
+        return i + 1, x + lax.psum(y, 'b')
+      return lax.while_loop(cond, body, (0, z))[1]
+    maps.xmap(f, axis_sizes=dict(a=2, b=10), out_axes=(['a']), in_axes={})(1.)
+
+  def test_while_loop_fixed_point_with_batched_pred_and_consts(self):
+    def f(i, x):
+      def cond(carry):
+        i, x = carry
+        return i < 5
+      def body(carry):
+        i, z = carry
+        # Close over const with batch dim = 1
+        return i + 1, z + x
+      return lax.while_loop(cond, body, (i, jnp.ones(3)))[1]
+    jax.vmap(f, in_axes=(0, 1))(jnp.arange(4), jnp.ones((3, 4)))
```

Comment:
Does one of those tests catch the breakage that the previous version has caused?

Reply:
Yes, this test with the closed over value does.